lib/matplotlib/font_manager.py

```diff
@@ -1356,17 +1356,20 @@ def _rebuild():
     if _fmcache:
         try:
             fontManager = json_load(_fmcache)
-            if (not hasattr(fontManager, '_version') or
-                fontManager._version != FontManager.__version__):
-                _rebuild()
-            else:
-                fontManager.default_size = None
-                _log.debug("Using fontManager instance from %s", _fmcache)
-        except TimeoutError:
-            raise
-        except Exception:
-            _rebuild()
-    else:
+        except FileNotFoundError:
+            _log.debug("No font cache found %s", _fmcache)
+        except json.JSONDecodeError:
+            _log.warning("Font cache parsing failed %s", _fmcache)
+
+        if (not hasattr(fontManager, '_version') or
```

Comment:
can't `fontManager` be `None` here?  If so, won't `fontManager._version != FontManager.__version__` give an AttributeError?

Reply:
```
>python -c "fontManager = None; print(not hasattr(fontManager, '_version') or fontManager._version != 123)"
True
```
However `(version mismatch)` log message is misleading in this situation

Reply:
Python is so unexpected sometimes...

I guess the log message is misleading, but you'll already have had one before with the real error, so prob fine?  Or you could just check that fontManager is not None before the rest of the conditional...

Reply:
I have updated the PR, now we should be fine.